flask_restplus_patched/namespace.py

```diff
@@ -46,11 +46,15 @@ def resolve_object(self, object_arg_name, resolver):
         ...        # user is a User instance here
         """
         def decorator(func):
-            @wraps(func)
             def wrapper(*args, **kwargs):
                 kwargs[object_arg_name] = resolver(kwargs)
                 return func(*args, **kwargs)
-            return wrapper
+            if isinstance(func, type):
```

Comment:
Please, swap the `if` block with the `wrapper` function as if it is a class, there is no need in the function at all.